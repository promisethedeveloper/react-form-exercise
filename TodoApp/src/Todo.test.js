import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

it("should render without crashing", () => {
	render(<Todo />);
});

it("should match snapshot", () => {
	const { asFragment } = render(<Todo />);
	expect(asFragment()).toMatchSnapshot();
});

it("should match snapshot when editing", () => {
	const { asFragment, getByText } = render(<Todo />);
	const editButton = getByText("Edit");
	fireEvent.click(editButton);
	expect(asFragment()).toMatchSnapshot();
});

it("should run the update function on form submission", () => {
	const updateMock = jest.fn();
	const { getByText } = render(<Todo update={updateMock} />);
	const editButton = getByText("Edit");
	fireEvent.click(editButton);
	const updateButton = getByText("Update!");
	fireEvent.click(updateButton);
	expect(updateMock).toHaveBeenCalled();
});

it("should run the delete function on button click", () => {
	const removeMock = jest.fn();
	const { getByText } = render(<Todo remove={removeMock} />);
	const deleteButton = getByText("X");
	fireEvent.click(deleteButton);
	expect(removeMock).toHaveBeenCalled();
});
