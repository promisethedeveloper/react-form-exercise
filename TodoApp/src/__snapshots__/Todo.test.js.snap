// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <li>
      default todo
    </li>
    <button>
      Edit
    </button>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot when editing 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        type="text"
        value="default todo"
      />
      <button>
        Update!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
